File: 1288.remove-covered-intervals.py
#
# @lc app=leetcode id=1288 lang=python3
#
# [1288] Remove Covered Intervals
#

# @lc code=start
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:

        intervals.sort(key=lambda x: x[0])
        res, prev = 1, intervals[0]

        for l, r in intervals[1:]:
            if prev[1] < r:
                if prev[0] != l:
                    res += 1
                prev = (l, r)

        return res


# @lc code=end

File: RustSolutions/src/1288.remove-covered-intervals.rs
/*
 * @lc app=leetcode id=1288 lang=rust
 *
 * [1288] Remove Covered Intervals
 */

// @lc code=start
impl Solution {
    pub fn remove_covered_intervals(mut intervals: Vec<Vec<i32>>) -> i32 {
        intervals.sort_unstable_by(|a, b| {
            if a[0] == b[0] {
                b[1].cmp(&a[1])
            } else {
                a[0].cmp(&b[0])
            }
        });

        for

    }
}
// @lc code=end

